instrumentation/http-url-connection/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/httpurlconnection/HttpUrlConnectionTest.java

```diff
@@ -0,0 +1,317 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.httpurlconnection;
+
+import static io.opentelemetry.api.trace.SpanKind.CLIENT;
+import static io.opentelemetry.api.trace.SpanKind.SERVER;
+import static io.opentelemetry.javaagent.instrumentation.httpurlconnection.StreamUtils.readLines;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+import static io.opentelemetry.semconv.trace.attributes.SemanticAttributes.NetTransportValues.IP_TCP;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpClientTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientInstrumentationExtension;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.assertj.core.api.AbstractLongAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class HttpUrlConnectionTest extends AbstractHttpClientTest<HttpURLConnection> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpClientInstrumentationExtension.forAgent();
+
+  static final List<String> RESPONSE = Collections.singletonList("Hello.");
+  static final int STATUS = 200;
+
+  @Override
+  public HttpURLConnection buildRequest(String method, URI uri, Map<String, String> headers)
+      throws Exception {
+    return (HttpURLConnection) uri.toURL().openConnection();
+  }
+
+  @Override
+  public int sendRequest(
+      HttpURLConnection connection, String method, URI uri, Map<String, String> headers)
+      throws Exception {
+    if (uri.toString().contains("/read-timeout")) {
+      connection.setReadTimeout((int) READ_TIMEOUT.toMillis());
+    }
+    try {
+      connection.setRequestMethod(method);
+      headers.forEach(connection::setRequestProperty);
+      connection.setRequestProperty("Connection", "close");
+      connection.setUseCaches(false);
```

Comment:
The groovy version of this class sets this property to `true`
```suggestion
      connection.setUseCaches(true);
```